dotnet/src/IntegrationTests/Plugins/PluginTests.cs

```diff
@@ -196,4 +196,35 @@ public class PluginTests
         // Act
         await plugin[functionName].InvokeAsync(kernel, arguments);
     }
+
+    [Theory]
+    [InlineData("Plugins/instacart-ai-plugin.json", "Instacart", "create")]
+    public async Task QueryInstacartPluginWithDynamicPayloadAsync(
+        string pluginFilePath,
+        string name,
+        string functionName)
+    {
+        // Arrange
+        using (var stream = System.IO.File.OpenRead(pluginFilePath))
+        {
+            var kernel = new Kernel();
+            using HttpClient httpClient = new();
+
+            // note that this plugin is not compliant according to the underlying validator in SK
+            var plugin = await kernel.ImportPluginFromOpenAIAsync(
+                name,
+                stream,
+                new OpenAIFunctionExecutionParameters(httpClient) { IgnoreNonCompliantErrors = true, EnableDynamicPayload = true }); ;
+
+            var arguments = new KernelArguments();
+            arguments["title"] = "Shopping List";
+            arguments["ingredients"] = new string[] { "Flour", "Sugar", "Eggs" };
+            arguments["instructions"] = new string[] { "Cream softened butter and granulated sugar", "Add eggs one at a time, mix well, and stir in vanilla extract", "Combine dry ingredients and mix" };
+            arguments["question"] = "what ingredients do I need to make chocolate cookies?";
+            arguments["partner_name"] = "OpenAI";
+
+            // Act
+            await plugin[functionName].InvokeAsync(kernel, arguments);
+        }
```

Comment:
Do we need some assertion in this test?

Reply:
I don't think so, unless you have a case where there's a failure, but the test does not fail. I run them quite often, and they always show issues if there are any.